File: Semester 1/FP/past work/18.12.2019_test/repository.py
from domain import *

class Repository:
    def __init__(self, Filename = None, Class = None):
        self.array = []
        self.filename = Filename
        self._class = Class
        self.load_file()

    def __getitem__(self, item):
        return self.array[item]

    def __len__(self):
        return len(self.array)

    def load_file(self):
        if self.filename == None:
            return
        f = open(self.filename, "r")
        lines = f.readlines()
        for i in lines:
            i = i.strip().split(",")
            obj = self._class(*i)
            self.store(obj)
        f.close()

    def store (self, obj):
        self.array.append(obj)

File: Semester 3/Advanced Programming Methods/FirstSeminar2/src/mvc/Repository.java
package mvc;

import mvc.Model.Model;

public class Repository {
    DynamicVector items;

    public Repository()
    {
        items = new DynamicVector();
    }

    public void store(Model item)
    {
        this.items.store(item);
    }

    public DynamicVector getItems() {
        return items;
    }
    
    public Model getItem(int i)
    {
        return this.items.getItem(i);
    }
}
